dir/fs.go

```diff
@@ -0,0 +1,136 @@
+package dir
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// rootedFS is io.FS implementation used in New.
+// root is the root of the file system tree passed to os.DirFS.
+type rootedFS struct {
+	fs.FS
+	Root string
+}
+
+// NewRootedFS create a rootedFS
+func NewRootedFS(fsys fs.FS, root string) rootedFS {
+	return rootedFS{fsys, root}
+}
+
+// UnionDirFS is a simple union directory file system interface
+type UnionDirFS interface {
+	fs.ReadDirFS
+	GetPath(elem ...string) (string, error)
+}
+
+// NewUnionDirFS create an unionDirFS on the current OS file system
+// by the named dirs
+func NewUnionDirFS(dirs ...string) unionDirFS {
```

Comment:
Never return a private type from an exported method.
```suggestion
func NewUnionDirFS(dirs ...string) UnionDirFS {
```

Reply:
Done.